File: Server/demo-core/src/main/java/com/ssafy/campinity/core/repository/curation/CurationRepository.java
package com.ssafy.campinity.core.repository.curation;

import com.ssafy.campinity.core.dto.CurationScrapListResDTO;
import com.ssafy.campinity.core.entity.curation.Curation;
import com.ssafy.campinity.core.entity.curation.CurationCategory;
import io.lettuce.core.dynamic.annotation.Param;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface CurationRepository extends JpaRepository<Curation, Integer> {
    @Query(value = "SELECT * FROM curation order by RAND() limit 3", nativeQuery = true)
    List<Curation> findRandomCuration();

    @Query(value = "SELECT c.id, c.created_at, c.updated_at, c.content, c.title, c.uuid, c.curation_category, c.first_image_path FROM curation AS c INNER JOIN curation_scrap ON curation_scrap.member_id = :memberId AND curation_scrap.curation_id = c.id", nativeQuery = true)
    List<Curation> findScrapList(@Param("memberId") int memberId);

    List<Curation> findByCurationCategory(CurationCategory curationCategory);

    Optional<Curation> findByUuid(UUID curationId);
}

File: Android/app/src/main/java/com/ssafy/campinity/domain/repository/CurationRepository.kt
package com.ssafy.campinity.domain.repository

import com.ssafy.campinity.data.remote.Resource
import com.ssafy.campinity.domain.entity.curation.CurationDetailItem
import com.ssafy.campinity.domain.entity.curation.CurationItem

interface CurationRepository {

    suspend fun getCurations(curationCategory: String): Resource<List<CurationItem>>

    suspend fun getCuration(curationId: String): Resource<CurationDetailItem>

    suspend fun scrapCuration(curationId: String): Resource<Boolean>

    suspend fun getScrapCurations(): Resource<List<CurationItem>>
}